build/tasks/build.js

```diff
@@ -6,350 +6,349 @@
 
 "use strict";
 
-module.exports = function( grunt ) {
-	const fs = require( "fs" );
-	const path = require( "path" );
-	const rollup = require( "rollup" );
-	const slimBuildFlags = require( "./lib/slim-build-flags" );
-	const rollupFileOverrides = require( "./lib/rollup-plugin-file-overrides" );
-	const srcFolder = path.resolve( `${ __dirname }/../../src` );
-	const read = function( fileName ) {
-		return grunt.file.read( `${ srcFolder }/${ fileName }` );
-	};
+const fs = require( "fs" );
+const path = require( "path" );
+const util = require( "util" );
+const exec = util.promisify( require( "child_process" ).exec );
+const rollup = require( "rollup" );
+const excludedFromSlim = require( "./lib/slim-exclude" );
+const rollupFileOverrides = require( "./lib/rollup-plugin-file-overrides" );
+const pkg = require( "../../package.json" );
+const minify = require( "./minify" );
+const getTimestamp = require( "./lib/getTimestamp" );
+const verifyNodeVersion = require( "./lib/verifyNodeVersion" );
+const srcFolder = path.resolve( __dirname, "../../src" );
+
+const minimum = [ "core" ];
+
+// Exclude specified modules if the module matching the key is removed
+const removeWith = {
+	ajax: [ "manipulation/_evalUrl", "deprecated/ajax-event-alias" ],
+	callbacks: [ "deferred" ],
+	css: [ "effects", "dimensions", "offset" ],
+	"css/showHide": [ "effects" ],
+	deferred: {
+		remove: [ "ajax", "effects", "queue", "core/ready" ],
+		include: [ "core/ready-no-deferred" ]
+	},
+	event: [ "deprecated/ajax-event-alias", "deprecated/event" ],
+	selector: [ "css/hiddenVisibleSelectors", "effects/animatedSelector" ]
+};
+
+async function read( filename ) {
+	return fs.promises.readFile( path.join( srcFolder, filename ), "utf8" );
+}
+
+// Remove the src folder and file extension
+// and ensure unix-style path separators
+function moduleName( filename ) {
+	return filename
+		.replace( `${srcFolder}${path.sep}`, "" )
+		.replace( /\.js$/, "" )
+		.split( path.sep )
+		.join( path.posix.sep );
+}
+
+async function readdirRecursive( dir, all = [] ) {
+	let files;
+	try {
+		files = await fs.promises.readdir( path.join( srcFolder, dir ), {
+			withFileTypes: true
+		} );
+	} catch ( e ) {
+		return all;
+	}
+	for ( const file of files ) {
+		if ( file.isDirectory() ) {
+			all = all.concat(
+				await readdirRecursive( path.join( dir, file.name ) )
+			);
+		} else {
+			all.push( moduleName( path.join( dir, file.name ) ) );
```

Comment:
This is modifying the input parameter, while in the `if` above we're not. I'd align here to avoid surprises; either always create a new one or push to the existing one.

I think I'd push for perf reasons as there may be a lot of file involved. That'd change the above recommendation from using array spread to `all.push( ...sth )`.